rules/TYPO312/v2/MigrateGeneralUtilityGPMergedRector.php

```diff
@@ -0,0 +1,171 @@
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v12\v2\MigrateGeneralUtilityGPMergedRector\MigrateGeneralUtilityGPMergedRectorTest
+ */
+final class MigrateGeneralUtilityGPMergedRector extends AbstractScopeAwareRector
+{
+    /**
+     * @readonly
+     */
+    private GeneralUtilitySuperGlobalsToPsr7ServerRequestFactory $globalsToPsr7ServerRequestFactory;
+
+    public function __construct(GeneralUtilitySuperGlobalsToPsr7ServerRequestFactory $globalsToPsr7ServerRequestFactory)
+    {
+        $this->globalsToPsr7ServerRequestFactory = $globalsToPsr7ServerRequestFactory;
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Migrate GeneralUtility::_GPmerged', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+$getMergedWithPost = GeneralUtility::_GPmerged('tx_scheduler');
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\ArrayUtility;
+
+$getMergedWithPost = $request->getQueryParams()['tx_scheduler'];
+ArrayUtility::mergeRecursiveWithOverrule($getMergedWithPost, $request->getParsedBody()['tx_scheduler']);
+CODE_SAMPLE
+            ),
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\ArrayUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+class MyActionController extends ActionController
+{
+    public function myMethod(): void
+    {
+        $getMergedWithPost = GeneralUtility::_GPmerged('tx_scheduler');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\ArrayUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+class MyActionController extends ActionController
+{
+    public function myMethod(): void
+    {
+        $getMergedWithPost = $this->request->getQueryParams()['tx_scheduler'];
+        ArrayUtility::mergeRecursiveWithOverrule($getMergedWithPost, $this->request->getParsedBody()['tx_scheduler']);
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Expression::class];
+    }
+
+    /**
+     * @param Expression $node
+     * @return Expression[]|null
+     */
+    public function refactorWithScope(Node $node, Scope $scope): ?array
+    {
+        $assignNode = $node->expr;
+        if (! $assignNode instanceof Assign) {
+            return null;
+        }
+
+        /** @var Variable $variable */
+        $variable = $assignNode->var;
+
+        $staticCall = $assignNode->expr;
+
+        if (! $staticCall instanceof StaticCall) {
+            return null;
+        }
+
+        if (! $this->nodeTypeResolver->isMethodStaticCallOrClassMethodObjectType(
+            $staticCall,
+            new ObjectType(GeneralUtility::class)
```

Comment:
for convention use string instead of ::class here

Reply:
we could write a custom rector rule for this to write class names as string ìf they start with `TYPO3` :smile: